### ISSUE TITLE
[sklearn#34489] RFC Licencing issues when contributors had sold their copyright to paid platforms

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#321]
**Mirrored from [scikit-learn#34489](https://github.com/scikit-learn/scikit-learn/issues/34489)**

Original title: RFC Licencing issues when contributors had sold their copyright to paid platforms
Original labels: RFC
Created: 2026-07-16T15:53:03Z

---

I have come across paid / bounty platforms that pay freelancers to open issues/PRs on OSS projects to test certain prompts and serve for finetuning AI tools. These PRs then get reviewed by maintainers and the resulting code as well as traces created during the process are used as training data for LLMs. Most certainly this data gets forward-sold to the companies developing the most popular AI models. This business model seems to be pretty popular these days.

The problem here is that with code submitted via a paid platform, contributors usually don't hold copyright anymore. They have sold it to the company that paid them. As a result, they are formally not allowed to grant their code to be used with a BSD licence.

What can we do about this?

I've opened PR #34490 to educate contributors. Apart from this:
- Can probabl pay for a lawyer who advises us on how to re-write scikit-learn's terms?
- We should also make sure probabl as an employer of the labs team grants employees to publish their work under scikit-learn's BSD licence.

The trouble a clean contributing policy could prevent is that years later, when maybe the copyright to PRs on scikit-learn gets sold and executed, we won't need to peel away code that was added by contributors that had actually never owned copyright of their work.

CC @GaelVaroquaux @glemaitre 

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/102 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
